### PR TITLE
SPECS: Update qemu to 5.2.0

### PR DESCRIPTION
**Pull request by russell-islam**
The current version 4.2.0 seems broken. Qemu can not
boot any image and gives the following error message.

"SeaBIOS (version rel-1.12.0-59-gc9ba5276e321-
prebuilt.qemu.orgqemu-system-x86_64:
/usr/src/mariner/BUILD/qemu-4.2.0/include/qemu/int128.h:22:
int128_get64: Assert. Aborted"

This patch updates the version to 5.2.0
which is the default in CentOS 8.

Signed-off-by: Muminul Islam <muislam@microsoft.com>

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- This version upgrade was necessary for Qemu based mariner host that is planned for customer use. Previous version 4.2.0 
- was buggy. This version is the default for Centos 8 that does not break the qemu functionality

###### Change Log  

- No CVE fixes in this PR. Kept some CVE fixes from the older version that are still applicable

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- N/A

###### Links to CVEs  
- N/A

###### Test Methodology

- Pipeline build id:  PR Build
- Built locally using  the following command
- sudo make build-packages -j$(nproc) CONFIG_FILE= REBUILD_TOOLS=y PACKAGE_IGNORE_LIST="openjdk8" RUN_CHECK=y  PACKAGE_BUILD_LIST="qemu-kvm" PACKAGE_REBUILD_LIST="qemu-kvm"
- sudo make go-tidy-all
- sudo make go-test-coverage

### CODE REVIEW

File: SPECS/qemu-kvm/qemu-kvm.spec [hunk 1]
```diff
@@ -197,19 +125,17 @@ fi
 
 %files
 %defattr(-,root,root)
-%license LICENSE
+#%license LICENSE
```
Comment (PawelWMS) [inline SPECS/qemu-kvm/qemu-kvm.spec:128]: Please don't comment out the `%license` macro.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
Please fill out the PR template and indicate how you've checked the package builds. Usually it's a local build of the package with the `RUN_CHECK=y` argument.

**Comment by russell-islam** [comment]
> Please fill out the PR template and indicate how you've checked the package builds. Usually it's a local build of the package with the `RUN_CHECK=y` argument. The tests should still be passing - you can check that by looking for the execution of the `%check` section in the build logs. The build itself will succeed if only the tests fail.

How do I test? using rpmbuild? Could you please share the details?

**Comment by PawelWMS** [comment]
> > Please fill out the PR template and indicate how you've checked the package builds. Usually it's a local build of the package with the `RUN_CHECK=y` argument. The tests should still be passing - you can check that by looking for the execution of the `%check` section in the build logs. The build itself will succeed if only the tests fail.
> 
> How do I test? using rpmbuild? Could you please share the details?

You just need to run the build with `RUN_CHECK=y`: `make build-packages [other_args] RUN_CHECK=y`. Then check the package build logs under `build/logs/pkggen/rpmbuilding` for the results from the `%check` section.

**Comment by russell-islam** [comment]
time="2021-09-20T13:15:38-07:00" level=debug msg="Interrupting job (failfast)."
time="2021-09-20T13:15:38-07:00" level=debug msg="RESULTS    : PASS 15 | ERROR 1 | FAIL 0 | SKIP 19 | WARN 0 | INTERRUPT 0 | CANCEL 4"
time="2021-09-20T13:15:38-07:00" level=debug msg="JOB TIME   : 41.97 s"
time="2021-09-20T13:15:38-07:00" level=debug msg="make[1]: *** [/usr/src/mariner/BUILD/qemu-5.2.0/tests/Makefile.include:125: check-acceptance] Error 9"
time="2021-09-20T13:15:38-07:00" level=debug msg="make[1]: Leaving directory '/usr/src/mariner/BUILD/qemu-5.2.0/build'"
time="2021-09-20T13:15:38-07:00" level=debug msg="make: *** [GNUmakefile:11: check-acceptance] Error 2"
time="2021-09-20T13:15:38-07:00" level=debug msg="One (or more) tests failed. Check log for further details"
time="2021-09-20T13:15:38-07:00" level=debug msg="+ '[' 2 -ne 0 ']'"
time="2021-09-20T13:15:38-07:00" level=debug msg="+ testsPassed=false"
time="2021-09-20T13:15:38-07:00" level=debug msg="+ '[' false = false ']'"
time="2021-09-20T13:15:38-07:00" level=debug msg="+ echo 'One (or more) tests failed. Check log for further details'"
time="2021-09-20T13:15:38-07:00" level=debug msg="+ exit 1"

**Comment by russell-islam** [comment]
make check-tcg failed
`
time="2021-09-20T13:14:46-07:00" level=debug msg="+ make check-tcg"
time="2021-09-20T13:14:46-07:00" level=debug msg="changing dir to build for make \"check-tcg\"..."
time="2021-09-20T13:14:46-07:00" level=debug msg="make[1]: Entering directory '/usr/src/mariner/BUILD/qemu-5.2.0/build'"
time="2021-09-20T13:14:46-07:00" level=debug msg="  BUILD   TCG tests for x86_64-softmmu"
time="2021-09-20T13:14:46-07:00" level=debug msg="  BUILD   x86_64-softmmu guest-tests with cc"
time="2021-09-20T13:14:46-07:00" level=debug msg="/usr/bin/ld: memory: warning: allocated section `.notes' not in segment"
time="2021-09-20T13:14:46-07:00" level=debug msg="/usr/bin/ld: hello: warning: allocated section `.notes' not in segment"
time="2021-09-20T13:14:46-07:00" level=debug msg="  RUN     TCG tests for x86_64-softmmu"
time="2021-09-20T13:14:46-07:00" level=debug msg="  BUILD   x86_64-softmmu guest-tests with cc"
time="2021-09-20T13:14:46-07:00" level=debug msg="  RUN     tests for x86_64"
time="2021-09-20T13:14:46-07:00" level=debug msg="  TEST    memory on x86_64"
time="2021-09-20T13:14:46-07:00" level=debug msg="qemu-system-x86_64: Error loading uncompressed kernel without PVH ELF Note"
time="2021-09-20T13:14:46-07:00" level=debug msg="make[3]: *** [../Makefile.target:162: run-memory] Error 1"
time="2021-09-20T13:14:46-07:00" level=debug msg="make[2]: *** [/usr/src/mariner/BUILD/qemu-5.2.0/tests/tcg/Makefile.qemu:85: run-guest-tests] Error 2"
time="2021-09-20T13:14:46-07:00" level=debug msg="make[1]: *** [/usr/src/mariner/BUILD/qemu-5.2.0/tests/Makefile.include:61: run-tcg-tests-x86_64-softmmu] Error 2"
time="2021-09-20T13:14:46-07:00" level=debug msg="make[1]: Leaving directory '/usr/src/mariner/BUILD/qemu-5.2.0/build'"
time="2021-09-20T13:14:46-07:00" level=debug msg="make: *** [GNUmakefile:11: check-tcg] Error 2"
`

**Comment by PawelWMS** [comment]
> # make check-tcg failed
> time="2021-09-20T13:14:46-07:00" level=debug msg="+ make check-tcg"
> time="2021-09-20T13:14:46-07:00" level=debug msg="changing dir to build for make "check-tcg"..."
> time="2021-09-20T13:14:46-07:00" level=debug msg="make[1]: Entering directory '/usr/src/mariner/BUILD/qemu-5.2.0/build'"
> time="2021-09-20T13:14:46-07:00" level=debug msg=" BUILD TCG tests for x86_64-softmmu"
> time="2021-09-20T13:14:46-07:00" level=debug msg=" BUILD x86_64-softmmu guest-tests with cc"
> time="2021-09-20T13:14:46-07:00" level=debug msg="/usr/bin/ld: memory: warning: allocated section `.notes' not in segment" time="2021-09-20T13:14:46-07:00" level=debug msg="/usr/bin/ld: hello: warning: allocated section `.notes' not in segment"
> time="2021-09-20T13:14:46-07:00" level=debug msg=" RUN TCG tests for x86_64-softmmu"
> time="2021-09-20T13:14:46-07:00" level=debug msg=" BUILD x86_64-softmmu guest-tests with cc"
> time="2021-09-20T13:14:46-07:00" level=debug msg=" RUN tests for x86_64"
> time="2021-09-20T13:14:46-07:00" level=debug msg=" TEST memory on x86_64"
> time="2021-09-20T13:14:46-07:00" level=debug msg="qemu-system-x86_64: Error loading uncompressed kernel without PVH ELF Note"
> time="2021-09-20T13:14:46-07:00" level=debug msg="make[3]: *** [../Makefile.target:162: run-memory] Error 1"
> time="2021-09-20T13:14:46-07:00" level=debug msg="make[2]: *** [/usr/src/mariner/BUILD/qemu-5.2.0/tests/tcg/Makefile.qemu:85: run-guest-tests] Error 2"
> time="2021-09-20T13:14:46-07:00" level=debug msg="make[1]: *** [/usr/src/mariner/BUILD/qemu-5.2.0/tests/Makefile.include:61: run-tcg-tests-x86_64-softmmu] Error 2"
> time="2021-09-20T13:14:46-07:00" level=debug msg="make[1]: Leaving directory '/usr/src/mariner/BUILD/qemu-5.2.0/build'"
> time="2021-09-20T13:14:46-07:00" level=debug msg="make: *** [GNUmakefile:11: check-tcg] Error 2"

It looks like the tests have failed. Please check if it's a regression. If not, fixing the failure is out of scope for this PR. If the tests used to pass, then please check what's wrong and fix them, if possible. If it's going to turn out that the test has no chances of passing due to how we want Mariner to be configured, or due to how our build environment is set up (and there's no easy way to remedy that), then please disable that particular test.

**Comment by russell-islam** [comment]
@PawelWMS fixed now.

**Review by PawelWMS — APPROVED** [review]
Changes look good to me. Approving, please only update the PR summary/template.

**Comment by russell-islam** [comment]
@PawelWMS  Any plan to merge this PR?

**Comment by jslobodzian** [comment]
QEMU was upgraded to a more recent version in main/2.0.  Closing this PR

**Pull request closed without merging**